### PR TITLE
The Cygwin/CI is broken.

### PR DESCRIPTION
**Pull request by clanmills**
Try using cmake to build instead of make.

### CODE REVIEW

**Comment by clanmills** [comment]
I think c:\Cyginw64\bin\bash is missing. The miserable thing does this:

```
echo %CMD%
rm -rf build; mkdir -p build; cd build;cmake .. -G 'Unix Makefiles' -DCMAKE_CXX_STANDARD=98 -DCMAKE_CXX_FLAGS=-Wno-deprecated; cmake --build . --config Release; cmake --build . --target python_tests --config Release
cd %APPVEYOR_BUILD_FOLDER%
if "%BUILD%"=="CYGWIN64" C:\cygwin64\bin\bash -c "%CMD%"
Command exited with code 127
```
It's Thanksgiving today.  Tomorrow I'll install a fresh Cygwin64 on Windows.  Maybe bash.exe has moved.

https://clanmills.com/states/Thanksgiving/

**Comment by clanmills** [comment]
And now we know.  CMake isn't installed.  Right.  I'll fix that.

**Comment by clanmills** [comment]
Another way to ensure you have a working CMake on Cygwin is to build it locally and publish a zip of the the build.  Pull down the zip (86mb) with curl and run `make install`.  cmake.exe will be in /usr/local/bin.  This will take 30 seconds.  Building from source takes about 90 minutes.

File: appveyor_mingw_cygwin.yml [hunk 1]
```diff
@@ -32,9 +32,14 @@ install:
...
+    
     - if "%BUILD%"=="CYGWIN64" C:\cygwin64\bin\bash -c "wget https://raw.githubusercontent.com/transcode-open/apt-cyg/master/apt-cyg ; chmod +x apt-cyg; mv apt-cyg /usr/local/bin"
-    - if "%BUILD%"=="CYGWIN64" C:\cygwin64\bin\bash -c "apt-cyg install cmake zlib-devel libexpat-devel libxml2-devel libxslt-devel python38 python38-pip python38-libxml2"
+    # Use change mirror used by apt-cyg.  #2003 cmake installed by default mirror is broken 
+    - if "%BUILD%"=="CYGWIN64" C:\cygwin64\bin\bash -c "apt-cyg mirror http://mirrors.kernel.org/sources.redhat.com/cygwin/"
+    - if "%BUILD%"=="CYGWIN64" C:\cygwin64\bin\bash -c "apt-cyg install cmake curl zlib-devel libexpat-devel libxml2-devel libxslt-devel python38 python38-pip python38-libxml2 tar"
     - if "%BUILD%"=="CYGWIN64" C:\cygwin64\bin\bash -c "/usr/bin/python3.8.exe -m pip install --upgrade pip"
+    # As a last resort, build CMake from source.  Caution:  This takes about 60 minutes.
+    # if "%BUILD%"=="CYGWIN64" C:\cygwin64\bin\bash -c "cd /tmp; curl -LO https://github.com/Kitware/CMake/releases/download/v3.22.0/cmake-3.22.0.tar.gz; tar xfz cmake-3.22.0.tar.gz; cd cmake-3.22.0 ; ./bootstrap ; make ; make install"
```
Comment (kmilos) [inline appveyor_mingw_cygwin.yml:42]: Remove? Should not really be an option anyway...

File: appveyor_mingw_cygwin.yml [hunk 2]
```diff
@@ -52,13 +57,12 @@ build_script:
     - cd %APPVEYOR_BUILD_FOLDER%
     - cmd: if "%BUILD%"=="MINGW64" C:\msys64\usr\bin\bash -c "%CMD%"
     - cmd: set "PATH=c:\cygwin64\usr\local\bin;c:\cygwin64\bin;c:\cygwin64\usr\bin;c:\cygwin64\usr\sbin;"
-    - cmd: set  CMD=rm -rf build
+    - cmd: set  CMD=cmake.exe --version
```
Comment (kmilos) [inline appveyor_mingw_cygwin.yml:60]: Remove since root cause was found?

**Review by kmilos — COMMENTED** [review]
I wonder if we're missing updating the repository info (for both MinGW and Cygwin) before installing packages in order to get the latest packages, or you want to use what Appveyor froze in time on purpose?

Also, for MinGW: might reduce network traffic and speed up the build a bit if installation is done in one `pacman -S` line instead of the for loop?

File: appveyor_mingw_cygwin.yml (hunk 1 above)
Comment (clanmills) [reply appveyor_mingw_cygwin.yml:42]: I think we should retain that comment.

If you want to polish the MinGW code, that's fine with me.

File: appveyor_mingw_cygwin.yml (hunk 2 above)
Comment (clanmills) [reply appveyor_mingw_cygwin.yml:60]: The `rm -rf build` code is pointless.   Reporting `cmake --version` is useful.

I've tried to submit a new PR and Git is going to fight me to the end.  Let's just get this working correctly and closed.

File: appveyor_mingw_cygwin.yml (hunk 1 above)
Comment (kmilos) [reply appveyor_mingw_cygwin.yml:42]: Ok, will do on this branch in a bit, thanks.

**Comment by clanmills** [comment]
Zee lights ease green.  Let's merge and close.

**Review by kmilos — APPROVED** [review]
Will make my suggestions in a new PR then...

**Pull request merged**